lib/optimize/CommonsChunkPlugin.js

```diff
@@ -46,8 +47,20 @@ The available options are:
 		const asyncOption = this.async;
 		const minSize = this.minSize;
 		const ident = this.ident;
+		const deepChildren = this.deepChildren;
 		compiler.plugin("this-compilation", (compilation) => {
 			compilation.plugin(["optimize-chunks", "optimize-extracted-chunks"], (chunks) => {
+				function pushUsedChunks(usedChunks, commonChunk, flags) {
+					(commonChunk.chunks || []).forEach((chunk) => {
+						if(flags.asyncOption || chunk.parents.every((parentChunk) => parentChunk === commonChunk || usedChunks.indexOf(parentChunk) >= 0)) {
```

Comment:
We can only use chunks if the commonChunk is the only parent. When recursively expanding all parents of the chunk (while ignoring circles) there must be only the commonChunk left. This condition must be ensured.

And we need a test for this. To create such a case you can create an additional entry point which also references the chunk which contains the common module (reference it via name, elsewise the optimizer will optimize it)